### ISSUE TITLE
phpMyAdmin version 5.1 exclusion plugin

### ISSUE THREAD

**Issue by azurit** [coreruleset/coreruleset#2394]
As URL structure is much different from version 5.1, new exclusion package/plugin needs to be created for versions above 5.1.

This exclusion rules plugin is ready and needs testing.

**Comment by azurit** [comment]
Hey @williamdes! Would you like to test this?
@lifeforms hm?

**Comment by lifeforms** [comment]
Sorry I don't have a phpMyAdmin 5 in production. I could set one up, but I have almost no time so that could take a while.

**Comment by azurit** [comment]
The biggest challenge of these exclusions was new URL structure which was introduced in version 5.1 of phpMyAdmin - all requests are routed through one script. The idea itself is good but the selection of the script name, `index.php`, made this difficult to implement without opening a huge hole it the firewall. Finally, i was able to come up with a solution which allows activation of this plugin without degrading the security of the whole web server.

Main idea is in the never-before-used-in-CRS mechanizm of dynamic URIs based on user configuration. First, user can define a subdirectory in which phpMyAdmin is running (and, of course, it must be really placed in that subdirectory*):
`setvar:'tx.phpmyadmin5-plugin_directory=/phpmyadmin5'"`

Then all exclusion rules are using this setting while matching the `REQUEST_FILENAME`:
`SecRule REQUEST_FILENAME "@endsWith %{tx.phpmyadmin5-plugin_directory}/index.php" \`

So, instead of binding to extremely common `/index.php` we are binding to quite unique `/phpmyadmin5/index.php`.

* User should create a redirection `/ -> /phpmyadmin5` so it's not needed to type `/phpmyadmin5` while accessing phpMyAdmin.

**Comment by azurit** [comment]
Hey @aus-coder, would you like to check this out?

**Comment by williamdes** [comment]
> Hey @williamdes! Would you like to test this? @lifeforms hm?

Hi !
You can use my nginx regex to determine all the allowed routes. It should be up to date and working.
See: https://github.com/sudo-bot/gh-deployer-container/blob/7f72d26f3a971cde2a5b3ee046b13c2980e97493/docker/nginx-default.conf#L29

**Comment by williamdes** [comment]
If you provide me some docker setup for testing the next rules I can validate them with some good testing :)

**Comment by theseion** [comment]
@azurit I'm trying to write up the instructions but I fail to find the plugin. I thought you already did all the work, didn't you? Is it just https://github.com/coreruleset/phpmyadmin-rule-exclusions-plugin?

**Comment by theseion** [comment]
Hi @williamdes. Thanks for offering to help. The following instructions should help you set up everything for testing (I'm not familiar with this plugin, so maybe wait for @azurit's response, unless you already know more than I do :) ).
1. Checkout branch v3.4/dev of the coreruleset repo
2. Copy the files from the [phpMyAdmin plugin](https://github.com/coreruleset/phpmyadmin-rule-exclusions-plugin) into the plugins directory of the coreruleset repo
3. Adapt the configuration of the [docker compose file](https://github.com/coreruleset/coreruleset/blob/v3.4/dev/tests/docker-compose.yml) to what you need:
  - BACKEND: attach your own backend if you want
  - MODSEC_RULE_ENGINE: disable detection only mode if that helps you test
  - PARANOIA: change the paranoia level to what you need
  - ...
4. Make sure your images are up to date:
  ```bash
  docker pull owasp/modsecurity-crs:apache
  docker pull owasp/modsecurity-crs:nginx
  ```
4. Run the containers:
  ```bash
  docker compose -f tests/docker-compose.yml up modsec2-apache
  ```

With https://github.com/coreruleset/modsecurity-crs-docker/pull/66 merged, the containers will automatically load the plugin configuration files from the plugins directory.

**Comment by azurit** [comment]
It's different from current phpmyadmin RE plugin. I didn't post it anywhere yet because i decided to integrate it into current phpmyadmin RE plugin (phpmyadmin version will be configurable).

Attaching phpmyadmin 5.1 RE plugin, it's work in progress and will be probably integrated into current phpmyadmin plugin (as i said above): [phpmyadmin5.tar.gz](https://github.com/coreruleset/coreruleset/files/8427882/phpmyadmin5.tar.gz)

**Comment by williamdes** [comment]
> This issue has been open 120 days with no activity. Remove the stale label or comment, or this will be closed in 14 days

Still on my todo list

**Comment by azurit** [comment]
Please look here:
https://github.com/coreruleset/phpmyadmin-rule-exclusions-plugin/pull/5

**Comment by fzipi** [comment]
Any news on this one? We would like to have this tested for the next major release!

**Comment by fzipi** [comment]
Removed from V4.

**Comment by williamdes** [comment]
> Any news on this one? We would like to have this tested for the next major release!

It's a bit complicated for me to test, the rules syntax are a bit complicated
Still on my infinite todo list
Very sorry for the very long long delay

**Comment by dune73** [comment]
I think @fzipi tries to clean up the todo list for v4 removing everything that is not essential. Given plugins are distributed separately from mainline CRS nothing
really changes outside of the v4 todo list.

**Comment by fzipi** [comment]
That's exactly what I'm doing. Unless you plan to have it working in the next weeks, this is not essential for v4.

**Comment by fzipi** [comment]
@azurit Sorry if my comment was a bit out of context. Going over the decisions in the last meeting we decided that anything plugin was not needed for v4 release. I just implemented that by removing from the milestone anything plugin related.

As this will not be released at the same time than v4, there is no need to tail the results from this effort to the v4 release.

**Comment by williamdes** [comment]
I can work onto it in the following days or weeks
Do you have documentation so I can try to write the ruleset from zero?
I need some way to gain knowledge on this blob of text filled with rules

**Comment by azurit** [comment]
Here is my latest work:
https://github.com/azurit/phpmyadmin-rule-exclusions-plugin/tree/v51support

I have already fully tested plugin for phpMyAdmin < 5.1 and, few days ago, started testing for phpMyAdmin > 5.1.

**Comment by azurit** [comment]
Done.

**Comment by williamdes** [comment]
Feel free to ping me to on PRs for phpMyAdmin or ask questions :)
